src/java/com/zimbra/cs/taglib/tag/LoginTag.java

```diff
@@ -164,9 +164,19 @@ public void doTag() throws JspException, IOException {
                 options.setAuthToken(mAuthToken);
                 options.setAuthAuthToken(true);
             } else {
+                String virtualHost = getVirtualHost(request);
+
+                if (mUsername != null && !mUsername.isEmpty() && mUsername.indexOf("@") != -1) {
+                    String usernameSplit[]= mUsername.split("@");
+
+                    // check if user domain matches current virtual host.
+                    if (!virtualHost.equals(usernameSplit[1])) {
+                        throw AuthFailedServiceException.AUTH_FAILED(usernameSplit[0], "", "Invalid virtual host");
```

Comment:
Don't we need to log the username including domain (=mUsername) and a virtual host specified in the request (=virtualHost)?